### PR TITLE
Cleanup anonymous namespace

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This PR cleans up the use of `namespace {` with the use of `namespace internal{`. This PR touches a lot of files but the changes are trivial.

`sort_indices` was moved outside of the anonymous namespace to the `stan::math` namespace as its used in some other files.

## Tests

/

## Side Effects

/

## Checklist

- [x] Math issue #1006 

- [x] Copyright holder: Rok Češnovar

- [x] the basic tests are passing

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
This is ready for review.

**Comment by syclik** [comment]
@rok-cesnovar, I got through about 1/4 of the way and it looks good. I'll just walk through some more of the changes before approving (unless someone wants to walk back from the end).

Thank you. I don't know how you're able to power through this stuff so quickly. There are a lot of sublte decisions you've made that are the right ones! (I was scratching my head for some of them for a second and then realized what was actually going on.)

**Comment by rok-cesnovar** [comment]
No rush, when you have the time. I have a few more of these simple code cleanup PRs lined up, but dont want to overcrowd by opening all of them up at once.

I am actually not sure on the `sort_indices` function. It would probably be best to hide it back to internal and add internal:: to the calls. But I am going to change that once you or anyone else gets through the rest of the changes.

It would probably be better if I split this up into more PRs, kinda like the type traits replacement. I realized that once I had everything finished. Sorry about that.

File: stan/math/prim/mat/fun/sort_indices.hpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ class index_comparator {
    * Construct an index comparator holding a reference
    * to the specified container.
    *
-   * @patam xs Container
```
Comment (bob-carpenter) [inline stan/math/prim/mat/fun/sort_indices.hpp]: Thanks for fixing the doc as a bonus!

File: stan/math/rev/mat/fun/log_sum_exp.hpp [hunk 2]
```diff
@@ -10,13 +10,13 @@
...
 
-namespace {
+namespace internal {
 
 // these function and the following class just translate
 // log_sum_exp for std::vector for Eigen::Matrix
 
 template <int R, int C>
-double log_sum_exp_as_double(const Eigen::Matrix<var, R, C>& x) {
+inline double log_sum_exp_as_double(const Eigen::Matrix<var, R, C>& x) {
```
Comment (bob-carpenter) [inline stan/math/rev/mat/fun/log_sum_exp.hpp]: Another good side fix.  Thanks.

File: stan/math/rev/mat/fun/sd.hpp [hunk 3]
```diff
@@ -12,12 +12,12 @@
...
 namespace math {
 
-namespace {  // anonymous
+namespace internal {
 
 // if x.size() = N, and x[i] = x[j] =
 // then lim sd(x) -> 0 [ d/dx[n] sd(x) ] = sqrt(N) / N
 
-var calc_sd(size_t size, const var* dtrs) {
+inline var calc_sd(size_t size, const var* dtrs) {
```
Comment (bob-carpenter) [inline stan/math/rev/mat/fun/sd.hpp]: Just want you to know I'm paying attention.

**Comment by syclik** [comment]
Thanks, @bob-carpenter!

And thanks @rok-cesnovar!

**Pull request merged**